ports/cheriot-rtos/machine_i2c.c

```diff
@@ -0,0 +1,129 @@
+#include <stdio.h>
+
+#include "py/obj.h"
+#include "py/runtime.h"
+#include "extmod/modmachine.h"
+
+typedef struct OpenTitanI2c open_titan_i2c_t;
+
+extern void i2c_setup(volatile open_titan_i2c_t *i2c, uint32_t freq_kHz);
+extern void i2c_send_address(volatile open_titan_i2c_t *i2c, uint8_t addr);
+extern bool i2c_blocking_read(volatile open_titan_i2c_t *i2c, uint8_t addr, uint8_t *buf, uint32_t len);
+extern void i2c_blocking_write(volatile open_titan_i2c_t *i2c, uint8_t addr, uint8_t *buf, uint32_t len, bool skipStop);
+
+extern bool  i2c_check_success(volatile open_titan_i2c_t *i2c);
+
+
+typedef struct _machine_i2c_obj_t {
+    mp_obj_base_t base;
+    volatile open_titan_i2c_t *i2c_block;
+} machine_i2c_obj_t;
+
+
+#define I2C_DEFAULT_FREQ (400000)
+
+static void machine_i2c_print(const mp_print_t *print, mp_obj_t self_in, mp_print_kind_t kind) {
+    machine_i2c_obj_t *self = (machine_i2c_obj_t *)MP_OBJ_TO_PTR(self_in);
+    const char *name;
+
+    switch ((uint32_t)self->i2c_block) {
+        case 2149580800:
+            name = "0";
+            break;
+        case 2149584896:
+            name = "1";
```

Comment:
Magic numbers should be `#define`d with readable names. Also, these look like addresses in which case they should be presented in hex notation.

Reply:
They are indeed addresses; this is essentially just a reverse lookup translating address into name. 
For now I will move them to `#define`s at the top of the file, but ideally, they should come from the sonata.json file. Origionally, I tried to use MMIO_CAPABILITY for this purpose, but it complained because that wasn't a literal the case can use. Suggestions on how to get around this?

Reply:
Under the hood an MMIO device named `exampledev` is a pair of linker symbols `__export_mem_exampledev` and `__export_mem_exampledev_end` defined in the linker script representing the lower and upper (respectively) bounds of the MMIO region associated with the device. `(ptraddr_t) &__export_mem_exampledev` (with `extern struct opaque __export_mem_exampledev;` declared previously) *might* work (although the capability will certainly be invalid, the address should be correct, not 100% certain though). Regardless that option is quite ugly and probably not ideal. You could store an identifier in the struct to be read back, or if it's really important write a script to generate #defines from a board JSON file but I don't think that's really necessary.

Reply:
I've gone through all the files and put in `#defines` for the relevant addresses